Extensions/read_more_now.js

```diff
@@ -1,211 +1,104 @@
-				post.find("a").each(function() {
-					var link = $(this);
-					var link_text = link.text().trim();
-					if ((link_text !== "Keep reading")
-                                           && (link_text !== "Weiterlesen")        //German
-                                           && (link_text !== "Afficher davantage") //French
-                                           && (link_text !== "Continua a leggere") //Italian
-                                           && (link_text !== "\u3055\u3089\u306B\u8AAD\u3080")         //Japanese
-                                           && (link_text !== "Okumaya devam et")   //Turkish
-                                           && (link_text !== "Seguir leyendo")     //Spanish
-                                           && (link_text !== "\u0427\u0438\u0442\u0430\u0442\u044C \u0434\u0430\u043B\u044C\u0448\u0435")      //Russian
-                                           && (link_text !== "Czytaj dalej")       //Polish
-                                           && (link_text !== "Continuar a ler")    //Portuguese (Portugal)
-                                           && (link_text !== "Continuar lendo")    //Portuguese (Brazil)
-                                           && (link_text !== "Lees verder")        //Dutch
-                                           && (link_text !== "\uB354 \uBCF4\uAE30")  //Korean
-                                           && (link_text !== "\u7EE7\u7EED\u9605\u8BFB") //Simplified Chinese
-                                           && (link_text !== "\u7E7C\u7E8C\u95B1\u8B80") /*Traditional Chinese (Hong Kong & Taiwan)*/) {
+			let cached_content = XKit.extensions.read_more_now.cache[postID];
+			if (cached_content === undefined) {
+				XKit.extensions.read_more_now.get_username(url)
+				.then(username => XKit.svc.indash_blog({
+					tumblelog_name_or_id: username,
+					post_id: postID,
+					limit: 1,
+					should_bypass_safemode: true,
+					should_bypass_tagfiltering: true
+				}))
+				.then(response => {
+					let responseData = response.json().response;
+					if (responseData.post_not_found_message !== undefined) {
 						return;
 					}
-					need_reflow = true;
 
-					if (/https?:\/\/[^.]+\.tumblr\.com\/post\/\d+/.test(link.attr("href"))) {
-						XKit.extensions.read_more_now.append_button_with_link(link.parent(), link.attr("href"));
-					} else {
-						// If url is not of the form
-						// http://something.tumblr.com/post/numbers/whatever it needs to be
-						// determined based on the post's author
-						var real_prefix = 'https://' + post.attr('data-tumblelog-name') + '.tumblr.com/';
-						var real_link = link.attr('href').replace(/https?:\/\/[^\/]+\//, real_prefix);
-						XKit.extensions.read_more_now.append_button_with_link(link.parent(), real_link);
-					}
+					let comment = responseData.posts[0].reblog.comment;
+					let readmore = comment.substring(comment.indexOf("[[MORE]]") + 8).replace(/\[\[MORE\]\]/g, "");
```

Comment:
why do both? shouldn't we just substring? if someone includes the [[MORE]] twice, that isn't really our problem, right?